packages/core-data/src/components/RecordDetailHeader.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import clsx from 'clsx';
+import React from 'react';
+import Icon from './Icon';
+import Button from './Button';
+import RecordDetailTitle from './RecordDetailTitle';
+import _ from 'underscore';
+import RecordDetailItem from './RecordDetailItem';
+import i18n from '../i18n/i18n';
+
+type Props = {
+  /**
+   * Content to be rendered as the blurb
+   */
+  children?: Node,
+
+  /**
+   * Class name to apply to the main div.
+   */
+  className?: string,
+
+  /**
+   * List of detail fields to be rendered above the blurb
+   */
+  detailItems?: Array<{ text: string, icon?: string, className?: string }>,
+
+  /**
+   * If a URL for a record detail page is provided, will render a button that links to it
+   */
+  detailPageUrl?: string,
+
+  /**
+   * The icon to display before the header.
+   */
+  icon?: string,
+
+  /**
+   * The text of the header.
+   */
+  title: string,
+
+  /**
+   * Additional classes to apply to the title element
+   */
+  titleClassName?: string
+};
+
+const RecordDetailHeader = (props: Props) => (
+  <div
+    className={clsx(
+      'flex',
+      'flex-col',
+      'gap-4',
+      'px-6',
+      'pt-6',
+      'pb-4',
+      props.className
+    )}
+  >
+    <RecordDetailTitle
+      text={props.title}
+      icon={props.icon}
+      className={props.titleClassName}
+    />
+    <ul>
```

Comment:
It might be a rare situation, but we shouldn't display an empty `<ul>` element when `props.detailItems` is undefined or empty.